### PR TITLE
Merge /collections/content and /fs/list endpoints

### LINKED ISSUE #164: Merge /collections/content/:collection_id and /collections/fs/list endpoints
As an effort to make the estuary API more clean and obvious, we could merge these two endpoints. They essentially do the same thing: list contents of a collection. The difference is in which information each shows. `/collections/fs/list` shows information about collection paths and subdirectories, whereas `/collections/content/:collection_id` has stats about pins and whatnot. I think that these two could be one endpoint with optional arguments to show directory structure or stats.

### PR DESCRIPTION
**Pull request by gmelodie**
Fixes #164

### CODE REVIEW

**Review by jlogelin — CHANGES_REQUESTED** [review]

File: handlers.go [hunk 1]
```diff
@@ -193,11 +193,10 @@ func (s *Server) ServeAPI() error {
 	cols.DELETE("/:coluuid", withUser(s.handleDeleteCollection))
 	cols.POST("/create", withUser(s.handleCreateCollection))
 	cols.POST("/add-content", withUser(s.handleAddContentsToCollection))
-	cols.GET("/content/:coluuid", withUser(s.handleGetCollectionContents))
+	cols.GET("/content", withUser(s.handleGetCollectionContents))
```
[the remark this answers is not in the document]
Comment (en0ma) [reply handlers.go:196]: that was why I suggested this https://github.com/application-research/estuary/issues/164#issuecomment-1151011818

**Review by jlogelin — COMMENTED** [review]
Friendly reminder: changes to the REST API requires a manual update to the swagger docs via `make generate-swagger`. (Hopefully, this will be added to the pipelines in the near future)

File: handlers.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (gmelodie) [reply handlers.go:196]: Yes there is. It's how most of the other API endpoints that accepts `coluuid` and `colpath` work.

[the remark this answers is not in the document]
Comment (en0ma) [reply handlers.go:196]: @gmelodie the `content/add` and `content/add-car` are different (content is the main resource, so collection can come in as query param). But for this, `collection` is the main resource, so it should not be a query param

[the remark this answers is not in the document]
Comment (gmelodie) [reply handlers.go:196]: I think the API is due some redesign, but for now the way our users seem to be thinking is that `coluuid` and `colpath` are query params (they don't seem to care what the main resource is). Maybe we could leave them as query params now and later we could do a v2 of the api that's properly RESTful

File: handlers.go [hunk 2]
```diff
@@ -3270,30 +3269,122 @@ func (s *Server) handleCommitCollection(c echo.Context, u *User) error {
...
+			}
+		}
+
+		var contentType CidType
+		contentType = File
+		if r.Type == util.Directory {
+			contentType = Dir
+		}
+		out = append(out, collectionListResponse{
+			// Name: filepath.Base(relp),
```
Comment (alvin-reyes) [inline handlers.go:3377]: Remove the // Name here

File: handlers.go [hunk 3]
```diff
@@ -3270,30 +3269,122 @@ func (s *Server) handleCommitCollection(c echo.Context, u *User) error {
...
+			Name:   r.Name,
+			Type:   contentType,
+			Size:   r.Size,
+			ContID: r.ID,
+			Cid:    &util.DbCID{r.Cid.CID},
+		})
+
+	}
+
+	return c.JSON(200, out)
```
Comment (alvin-reyes) [inline handlers.go:3387]: use `http.StatusOk`

File: handlers.go [hunk 4]
```diff
@@ -3270,30 +3269,122 @@ func (s *Server) handleCommitCollection(c echo.Context, u *User) error {
...
-		Scan(&contents).Error; err != nil {
+		Select("contents.*, collection_refs.path as path"). // TODO: change content.name to content.filename
+		Scan(&refs).Error; err != nil {
 		return err
 	}
 
-	return c.JSON(200, contents)
+	dir := c.QueryParam("colpath")
+	if dir == "" {
+		return c.JSON(200, refs)
```
Comment (alvin-reyes) [inline handlers.go:3299]: use `http.StatusOk`

File: handlers.go [hunk 5]
```diff
@@ -3270,30 +3269,122 @@ func (s *Server) handleCommitCollection(c echo.Context, u *User) error {
...
+	for _, r := range refs {
+		if r.Path == "" || r.Name == "" {
+			continue
+		}
+
+		path := r.Path
+
+		relp, err := filepath.Rel(dir, path)
+		if err != nil {
+			return err
```
Comment (alvin-reyes) [inline handlers.go:3316]: Let's return the proper http status with a message.

**Comment by en0ma** [comment]
@gmelodie please address conflict and @alvin-reyes comments so this can be approved and merged

**Review by alvin-reyes — APPROVED** [review]
The docs/docs.go, and swagger.json/yaml will be auto-generated by build process once it's on the main branch.

LGTM.

**Pull request merged**